pycalphad/core/calculate.py

```diff
@@ -312,18 +313,6 @@ def calculate(dbf, comps, phases, mode=None, output='GM', fake_points=False, bro
         raise ValueError('The \'points\' keyword argument must be specified if broadcast=False is also given.')
     nonvacant_components = [x for x in sorted(comps) if x.number_of_atoms > 0]
 
-    # Convert keyword strings to proper state variable objects
-    # If we don't do this, sympy will get confused during substitution
-    statevar_dict = dict((v.StateVariable(key), unpack_condition(value)) for (key, value) in kwargs.items())
-    # XXX: CompiledModel assumes P, T are the only state variables
```

Comment:
Now that the `raise` has been re-enabled in callables, the user gets an error if the state variable is unspecified but actually required for the model. (You can drop `P` if it doesn't show up in the model.)

I think this is an acceptable design for now, because I'm not sure yet how exactly NP conditions will work with this. (Is it `calculate` or `build_callables`'s job to handle unspecified state variables, or is it handled further up the call stack?) It needs a test, but I'd prefer to wait until NP conditions so I know what the test should actually be.